src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -177,6 +223,45 @@ public class PatientEverythingService : IPatientEverythingService
             return new SearchResult(searchResultEntries, searchResult.ContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
         }
 
+        private void StoreSeeAlsoLinkInformationInToken(string resourceId, EverythingOperationContinuationToken token, IReadOnlyCollection<SearchResultEntry> searchResultEntries)
+        {
+            SearchResultEntry patientResource = searchResultEntries.FirstOrDefault(s => string.Equals(s.Resource.ResourceTypeName, ResourceType.Patient.ToString(), StringComparison.Ordinal));
+
+            if (patientResource.Resource != null)
+            {
+                ResourceElement element = _resourceDeserializer.Deserialize(patientResource.Resource);
+                Patient patient = element.ToPoco<Patient>();
+
+                List<Patient.LinkComponent> links = patient.Link;
+
+                if (links != null)
+                {
+                    foreach (Patient.LinkComponent link in links)
+                    {
+                        ReferenceSearchValue referenceSearchValue = _referenceSearchValueParser.Parse(link.Other.Reference);
+
+                        // If the link does not point back to the current patient
+                        if (!string.Equals(referenceSearchValue.ResourceId, resourceId, StringComparison.Ordinal))
+                        {
+                            if (link.Type == Patient.LinkType.ReplacedBy)
+                            {
+                                throw new EverythingOperationException(string.Format(Core.Resources.EverythingOperationResourceIrrelevant, resourceId, referenceSearchValue.ResourceId), HttpStatusCode.BadRequest);
+                            }
+
+                            if (link.Type == Patient.LinkType.Seealso)
+                            {
+                                // Links can be of type Patient or RelatedPerson - only attempt to run the $everything operation on Patient resources
```

Comment:
This will change in [AB#85142](https://microsofthealth.visualstudio.com/Health/_workitems/edit/85142)